source/backend/discovery/src/lib/apiClient/index.js

```diff
@@ -79,25 +86,124 @@ function updateCrawledAccounts(appSync) {
 
 function addCrawledAccounts(appSync) {
     return async accounts => {
-        const crawledAccounts = accounts.map(({regions, isIamRoleDeployed, lastCrawled, credentials, ...props}) => {
+        return Promise.resolve(accounts)
+            // we must ensure we do not persist any temporary credentials to the db
+            .then(R.map(R.omit(['credentials', 'toDelete'])))
+            .then(R.map(({regions, isIamRoleDeployed, lastCrawled, ...props}) => {
+                return {
+                    ...props,
+                    isIamRoleDeployed,
+                    regions: regions.map(name => ({name})),
+                    lastCrawled: isIamRoleDeployed ? new Date().toISOString() : lastCrawled
+                }
+            }))
+            .then(appSync.addAccounts);
+    }
+}
+
+async function getOrgAccounts(
+    {ec2Client, organizationsClient, configClient}, appSyncClient, {configAggregator}
+) {
+    const [{Arn: managementAccountArn}, dbAccounts, orgAccounts, {OrganizationAggregationSource}, regions] = await Promise.all([
+        organizationsClient.getRootAccount(),
+        appSyncClient.getAccounts(),
+        organizationsClient.getAllAccounts(),
+        configClient.getConfigAggregator(configAggregator),
+        ec2Client.getAllRegions()
+    ]);
+
+    const dbAccountsMap = new Map(dbAccounts.map(x => [x.accountId, x]));
+    const orgAccountsMap = new Map(orgAccounts.map(x => [x.Id, x]));
+
+    const deletedAccounts = dbAccounts.reduce((acc, account) => {
+        const {accountId} = account;
+        if(dbAccountsMap.has(accountId) && !orgAccountsMap.has(accountId)) {
+            acc.push({...account, toDelete: true});
+        }
+        return acc;
+    }, []);
+
+    const managementAccountId = parseArn(managementAccountArn).accountId;
+
+    return orgAccounts
+        .map(({Id, Name: name, Arn}) => {
+            const [, organizationId] = parseArn(Arn).resource.split('/');
+            const lastCrawled = dbAccountsMap.get(Id)?.lastCrawled;
```

Comment:
What does last crawled contain?

Reply:
The timestamp of the last time the discovery process successfully discovered the account.